modules/runner/src/testFixtures/java/org/apache/ignite/internal/table/NodeUtils.java

```diff
@@ -49,16 +50,16 @@ public class NodeUtils {
      *
      * @param nodes Nodes collection.
      * @param groupId Group id.
-     * @param preferablePrimary Primary replica name which is preferred for being primary or {@code null}.
+     * @param preferablePrimaryFilter Primary replica preferable nodes filter, accepts the node consistent id.
      * @return New primary replica name.
      * @throws InterruptedException If failed.
      */
     public static String transferPrimary(
             Collection<IgniteImpl> nodes,
             ReplicationGroupId groupId,
-            @Nullable String preferablePrimary
+            @Nullable Predicate<String> preferablePrimaryFilter
```

Comment:
Is it all about ::contains support? Don't you think that it's a bit too complicated?
Anyway, I'd definitely preserve original method as sugar.

Reply:
I added a method with 2 parameters (nodes, group id) and returned the old one.